### ISSUE TITLE
BUG: Admissions clerk is unable to add in an email for a Petitioner with an international address - getting a 400 error.

### ISSUE THREAD

**Issue by ttlenard** [ustaxcourt/ef-cms#8117]
**Describe the Bug**
When a petitioner has an international address, and then requests eAccess, a 400 error is thrown when an admissions clerk attempts to save their email so that they can have eAccess.  Based off of the logs, it appears that the error has something to do with DAWSON looking for a US state in the state field, rather than an international one?  

Note - this error does not occur on change of address (going from a US address to International), Admissions clerk only receives the error when attempting to add in an email for the petitioner.

Note: I was attempting to see if I could do a work around by removing the data in the state/province/region field, and adding it to the address line 2 field. Then adding in the petitioner email so it wouldn't error out, and then switching the address back to how it was (yes, this results in some unnecessary Notice of change of addresses), but then there was some really funky behavior where the petitioner had eAccess, but it didn't register and showed "No email provided".  Not a solid work around in my opinion, but I tried! See steps below.

**Business Impact/Reason for Severity**

**In which environment did you see this bug?**
Prod, Mig, Test

**Who were you logged in as?**
Admissions clerk

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
This was reported by a court admissions clerk while using the application

**To Reproduce**
1. Log in as a admissions clerk
2. Go to a case where there is a petitioner that has paper service (and no counsel). I used 222-21, 666-21, and 12008-21 while testing this.
3. Click on case information tab, then on Parties
3. Click on Edit for the petitioner contact information
4. Change their address so that they have an international address. This is the information that I input, based off of the real data in 12008-21.

- Country - International
- Country Name - Ireland
- Address line 1 - 18 Castlegate Way
- City - Co Dublin
- State/Province/Region - Adamstown, Lucan
- Zip Code - K78 CH24

5. Click Save
6. Check the docket record, A notice of change of address will be generated
7. Go back to Case Information and click on Parties
8. Click on Edit to edit the petitioners info
9. Add an email address to the Petitioner (I used my own email so that I can verify it)
10. Click Save. Notice that you receive a 400 error.
11. If you review the logs at this step (see below in screen grabs), you will see that you get an error that the state must include US states and Territories only.

**Expected Behavior**

Admissions clerk should be able to add in an email for international petitioners without an error message

**Actual Behavior**
A clear and concise description of what actually happened.
Admissions clerk received 400 error when attempting to add in an email address for a petitioner that has an international address

**Screenshots**

Error message admissions clerk gets when attempting to add in an email address for petitioner when the state/region/province field has Adamstown, Lucan in it.
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/1237ae1a-a1e1-4496-86ac-1f8c6f211351)

Log of the error received:
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/965a8377-3594-4e61-bf73-6d6a763242e4)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Ready for Bugs(Created 10-4-21)
Definition used: A failure or flaw in the system which produces an incorrect or undesired result that deviates from the expected result or behavior. (Note: Expected results are use cases that have been documented in past user stories as acceptance criteria and test cases, and do not include strange behavior unrelated to use cases.)

The following criteria must be met in order for the development team to begin work on the bug.

The bug must:
 - Be focused on solving a user problem
 - Contain data for all fields in the bug template, so the team can pick it up and begin working immediately

Process: If the unexpected results are new use cases that have been identified, but not yet built, new acceptance criteria and test cases should be captured in a new user story and prioritized by the product owner. 

If the Court is not able to reproduce the bug, add the “Unable to reproduce” tag. This will provide visibility into the type of support that may be needed by the Court. In the event that the Court cannot reproduce the bug, the Court will work with Flexion to communicate what type of troubleshooting help may be needed.

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [ ] Automated test scripts have been written
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Deployed to the Court's test environment

**Comment by ttlenard** [comment]
Some additional feedback to summarize this thing: 

I attempted one more thing this morning to see if I could get a good workaround, but it didn't happen.  I cannot get a petitioner with an international address to show up as having eAccess on the contact info card.  (They actually DO get it, but DAWSON doesn't display their email address, and it appears that they don't have eAccess, when they actually do.  Also, in order for them to get the eAccess, we have to tweak their physical address so that it is incorrectly entered by putting the state/province/region in a different address line which generates a Notice of Change of Address (this has an incorrect address), and then we have to switch it back to the correct format, which then generates ANOTHER change of address.

**Comment by ttlenard** [comment]
Another note: I also tested this to see what happens if a petitioner that lives internationally efiles their case. All seemed to work just fine.  I entered in the same exact address as above, then as the petitioner, submitted my case.  Petitions clerk served the case to IRS, and then as the International Petitioner, I changed my email address.  All went smooth. Seems this is only happening in the above scenario with admissions clerk workflow?
